lib/rules/no-misleading-character-class.js

```diff
@@ -320,12 +323,13 @@ module.exports = {
                             } else {
                                 foundKindMatches.set(kind, [...findCharacterSequences[kind](chars)]);
                             }
-
                         }
                     }
                 }
             });
 
+            const generateReportLocation = createReportLocationGenerator(node);
```

Comment:
[Refactor] I tried refactoring this locally to not generate a new report location generator within each `verify`. I.e. I deleted the `function createReportLocationGenerator(node) {` and corresponding `}`, moved the `// Only literals and ...` comment & `if` inside `generateReportLocation`, and had it called like:

```ts
const locs = generateReportLocation(matches, node);
```

...and no unit tests failed. Which means one or two of the following must be true:

* There is logical benefit that isn't captured by unit tests. Resolution: more unit tests!
* There is performance benefit that I missed. Resolution: it should probably be commented in the code, I think.
* There is no logical or performance benefit to this, so it should be simplified.

I'm not confident I know which it is. 😄

Reply:
> I'm not confident I know which it is. 😄

Now that you mention it, I'm not confident either. Let me check again.

Reply:
It's correct that the code should have been simplified 👍🏻 I wrote that `createReportLocationGenerator` function as a way to encapsulate the logic that chooses the implementation for generating report locations, but I realized that the default implementation wasn't used any more. After removed the unused part, the body of the function was just returning another function, so I've inlined that code into `verify`. It's a little simpler now than before the PR.